parser.py

```diff
@@ -255,51 +33,12 @@ def __repr__(self):
         return f"<QasmProgram: version={self.version}>"
 
 
-
-# TODO: make separate declaration class for register declarations;
-# distinguish from QDeclarations
-
-class ClassicalRegister(Declaration):
-
-    def __init__(self, name, size):
-        super().__init__(decl_type="creg", id_=name, size=size)
-
-    def __repr__(self):
-        return f"creg {self.kwargs['id_']}[{self.kwargs['size']}];"
-
-
-class QuantumRegister(Declaration):
-
-    def __init__(self, name, size):
-        super().__init__(decl_type="qreg", id_=name, size=size)
-
-    def __repr__(self):
-        return f"qreg {self.kwargs['id_']}[{self.kwargs['size']}];"
-
-
-class QDeclaration(Declaration):
-    pass
-
-
-class GateDeclaration(QDeclaration):
-
-    def __init__(self, op):
-        super().__init__(decl_type="gate", op=op)
-
-    def declaration_str(self):
-        return f"gate {self.kwargs['op']}"[:-1]
-
-
-class OperatorDeclaration(QDeclaration):
-    def __init__(self, op):
-        super().__init__(decl_type="operator", op=op)
-
-    def declaration_str(self):
-        return f"operator {self.kwargs['op']}"[:-1]
-
-
 class QASMToIRTransformer(Transformer):
+    """Transformer for processing the Lark parse tree.
 
+    Transformers visit each node of the tree, and run the appropriate method on it according to the node's data.
+    All method names mirror the corresponding symbols from the grammar.
```

Comment:
We may as well update the symbols from the grammar, it is very dense and hard to read (e.g., things like `uop` and `goplist` should have better names when lexing (such as `unary_op`), it's a big overhead for new developers)